hedera-node/hedera-evm/src/main/java/com/hedera/node/app/service/evm/store/contracts/AbstractEvmStackedLedgerUpdater.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.node.app.service.evm.store.contracts;
+
+import com.hedera.node.app.service.evm.accounts.AccountAccessor;
+import com.hedera.node.app.service.evm.store.models.UpdatedHederaEvmAccount;
+import org.hyperledger.besu.datatypes.Address;
+import org.hyperledger.besu.evm.account.Account;
+import org.hyperledger.besu.evm.worldstate.WorldView;
+
+public class AbstractEvmStackedLedgerUpdater<W extends WorldView, A extends Account>
+        extends AbstractLedgerEvmWorldUpdater<
+                AbstractLedgerEvmWorldUpdater<W, A>, UpdatedHederaEvmAccount<A>> {
+
+    protected AbstractEvmStackedLedgerUpdater(
+            final AbstractLedgerEvmWorldUpdater<W, A> world,
+            final AccountAccessor accountAccessor,
+            final HederaEvmEntityAccess entityAccess) {
+        super(world, accountAccessor, entityAccess);
+    }
+
+    @Override
+    public UpdatedHederaEvmAccount<A> getForMutation(Address address) {
+        final var wrapped = wrappedWorldView();
+        final A account = wrapped.getForMutation(address);
```

Comment:
Since we have added updatedAccounts map, probably we can use the optimisation strategy:

```
final var wrappedMutable = wrapped.updatedAccounts.get(address);
        if (wrappedMutable != null) {
            return wrappedMutable;
        }
```

Reply:
TODO - to be resolved in a future PR